src/AutoRest.CSharp/MgmtTest/Generation/MgmtBaseTestWriter.cs

```diff
@@ -0,0 +1,713 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using AutoRest.CSharp.Output.Models.Types;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Generation.Writers;
+using AutoRest.CSharp.Mgmt.AutoRest;
+using AutoRest.CSharp.Input;
+using AutoRest.CSharp.Mgmt.Decorator;
+using AutoRest.CSharp.Mgmt.Output;
+using AutoRest.CSharp.Output.Models.Shared;
+using AutoRest.CSharp.Utilities;
+using AutoRest.CSharp.Output.Builders;
+using AutoRest.CSharp.Output.Models.Serialization.Json;
+using System.Text.Json;
+using AutoRest.CSharp.Mgmt.Generation;
+using AutoRest.CSharp.Mgmt.Models;
+using AutoRest.CSharp.Output.Models.Requests;
+using System.Text.Json.Serialization;
+using Azure.Core.Serialization;
+
+namespace AutoRest.CSharp.MgmtTest.Generation
+{
+    internal partial class MgmtBaseTestWriter: MgmtClientBaseWriter
+    {
+        public static CodeWriter _tagsWriter = new CodeWriter();
+        public static HashSet<string>  variableNames = new HashSet<string>();
+
+        public MgmtBaseTestWriter(CodeWriter writer, MgmtTypeProvider provider, BuildContext<MgmtOutputLibrary> context) : base(writer, provider, context)
+        {
+        }
+
+        public void WriteTestDecorator()
+        {
+            _writer.Line($"[RecordedTest]");
+
+            var testsConfiguration = Context.Configuration.MgmtConfiguration.TestModeler;
+            System.Text.Json.JsonElement ignoreReason = default;
+            if (testsConfiguration is not null && testsConfiguration.IsObject() && testsConfiguration?.TryGetProperty("ignore-reason", out ignoreReason) == true)
+            {
+                _writer.UseNamespace("NUnit.Framework");
+                _writer.Line($"[Ignore(\"{ignoreReason.GetString()}\")]");
+            }
+        }
+
+        public static object? ConvertToStringDictionary(object dict)
+        {
+            if (dict is null)
+            {
+                return dict;
+            }
+            Type t = dict.GetType();
+            bool isDict = t.IsGenericType && t.GetGenericTypeDefinition() == typeof(Dictionary<,>);
+            if (!isDict)
+            {
+                return dict;
+            }
+            var ret = new Dictionary<string, object?>();
+            foreach (KeyValuePair<object, object> entry in (IEnumerable< KeyValuePair<object, object>>)dict)
+            {
+                ret.Add(entry.Key.ToString()!, ConvertToStringDictionary(entry.Value));
+            }
+            return ret;
+        }
```

Comment:
It's used to serialize example value of type 'any' in example: https://github.com/Azure/autorest.csharp/blob/5669bdaacbd2c9f603b41226c3670c6e20379386/src/AutoRest.CSharp/MgmtTest/Generation/MgmtBaseTestWriter.cs#L488

It return 'object?' since the 'JsonSerializer.Serialize()' can handle this type of parameter

Reply:
Redesigned into this function: https://github.com/Azure/autorest.csharp/blob/3deaeff2c8a814fbcf9159ef117288cecdf67343/src/AutoRest.CSharp/MgmtTest/Generation/MgmtBaseTestWriter.cs#L343